### ISSUE TITLE
Panic on f32-64::INFINITE/f32-64::NEG_INFINITE/f32-64::NAN output

### ISSUE THREAD

**Issue by james-womack** [async-graphql/async-graphql#735]
## Expected Behavior

When returning infinite/NaN floating-point values from a GraphQL Query response, the server should return a Server Error or equivalent to the client making a GraphQL request (JSON cannot represent an infinite number).

## Actual Behavior

The `async-graphql` parser causes a panic and unexpectedly kills the server. The site of where the panic happens is: https://github.com/async-graphql/async-graphql/blob/master/src/types/external/floats.rs#L21 on the `unwrap` call.

## Steps to Reproduce the Problem

1. Create a `SimpleObject` that contains one or more floating point fields. E.g.

```rust
#[derive(InputObject, Serialize, Deserialize)]
pub struct IMUFrame {
    pub quaternion_1: f32
}
```

2. Create a `Query` resolver that returns the said object, with an infinite value:

```rust
 #[graphql(name="getImuData")]
    async fn get_imu_data(&self) -> Result<IMUFrame> {
        Ok(IMUFrame { quaternion_1: f32::INFINITE })
    }
```

3. Run a query against the built schema with `query { getImuData { quaternion_1 } }` to cause a panic.

## Specifications

- Version: ^2.10
- Platform: Linux (armv7)
- Subsystem: N/A?

**Comment by sunli829** [comment]
Fixed in `v3.0.11` 🙂

**Comment by james-womack** [comment]
Thanks!
